servers/visual_server.cpp

```diff
@@ -364,6 +368,10 @@ Vector2 VisualServer::tangent_to_oct(const Vector3 v, const float sign, const bo
 // Convert Octohedron-mapped normalized vector back to Cartesian
 // Assumes normalized format (elements of v within range [-1, 1])
 Vector3 VisualServer::oct_to_norm(const Vector2 v) {
+	if (Math::is_equal_approx(v.length(), 0)) {
```

Comment:
Better:

```suggestion
	if (Math::is_zero_approx(v.length())) {
```

Even better (since we don't need to compute the length):

```suggestion
	if (v.is_equal_approx(Vector3())) {
```

Reply:
will make the change

Reply:
Just to clarify, the octahedral compressed Vector2 itself does not have to be a normalized vector itself, it just has to have normalized elements  (within range [-1, 1] ) 

and clayjohn's suggestion to warn when the passed in octahedral compressed vector has elements outside the range [-1, 1] is probably what I would prefer here

Reply:
OK I am alright with that, but we do need to deal with the problem that the code generates NaN values with a zero input so we can solve that another way.

So in theory having a 0,0,0 input will result in NaN consistently, which is resulting in undefined behavior so we'd be relying on that.

So we should really solve this so the division by zero cannot happen, then in theory any input is totally okay.